compiler/rustc_middle/src/ty/instance.rs

```diff
@@ -19,14 +19,14 @@ use std::fmt;
 /// simply couples a potentially generic `InstanceDef` with some args, and codegen and const eval
 /// will do all required substitution as they run.
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug, TyEncodable, TyDecodable)]
-#[derive(HashStable, Lift, TypeFoldable, TypeVisitable)]
+#[derive(HashStable, TypeFoldable, TypeVisitable)]
```

Comment:
This one is used by cg_clif's jit mode: https://github.com/bjorn3/rustc_codegen_cranelift/blob/d72f7109ede688200c4ffdc0453457a66552f47b/src/driver/jit.rs#L256

Reply:
I have reinstated it, along with the ones for `InstanceDef` and `Option`, which were also required.

rustc_codegen_cranelift's `Cargo.toml` says the `jit` feature is "not ready to be enabled when compiling as part of rustc", which is a bit annoying when making this kind of change. Is that still true? If so, what is required to enable it?

Reply:
You can edit `Cargo.toml` to add `jit` to the default features for testing. It is indeed still not ready to be enabled by default. `std::env::args()` is broken on windows, macos and linux with glibc. (on these targets the argv passed to the main function is ignored), the jitted code doesn't run on the main thread which breaks everything using winit and in general it is slower due to incr comp being broken as an exit from the jitted main function will immediately exit the rustc process and because cranelift-jit doesn't support serializing to a cachable object file. It also requires all dependencies to be available as dylib.

Reply:
Would it be possible to add `jit` to the features within rustc just so it gets compiled, even if it's not tested/run in any way? That would be enough to avoid the above problem.

Reply:
Should be possible to add it to check builds.

Reply:
Cool. Doesn't need to be in this PR, though, which I think is ready to be merged.